integration-tests/pom.xml

```diff
@@ -7,7 +7,7 @@
     <parent>
         <groupId>oracle.kubernetes</groupId>
         <artifactId>operator-parent</artifactId>
-        <version>3.3.4</version>
+        <version>4.0.0</version>
```

Comment:
As above, I'll look at this too... Part of the reason that I've not used the Maven plugin is that we have a bunch of references that aren't easily driven from the Maven version, such as documentation, Helm chart default values, and samples, etc. Therefore, I'd just found it easier to do a grep and replace. I did find an easier way to update the references in the documentation with a change to just one file, but that is still separate from the Maven version.

Reply:
I'm thinking that now that we are running a process to publish the docs, we could incorporate the maven version at that time. It would mean, of course, that we would only update them when we publish the release, rather than on every push to main.

Reply:
The maven release plugin runs steps in the project poms based on profiles. We could certainly incorporate grep-type replacements as part of that. We could also include the Hugo generation, and move all of the web page updating into that step. It would be much easier to test than a GitHub action.

Reply:
I'm open to using the release plugin but that needs to be more thought through and, therefore, outside the scope of this PR. For instance, the Hugo generation needs to be separate or at least separable from the release process since we often correct errors or update the documentation outside of the release process. I do strongly support automating as much of the release process as possible.